packages/block-editor/src/components/block-list/index.js

```diff
@@ -121,73 +136,72 @@ class BlockList extends Component {
 			return;
 		}
 
-		const boundaries = this.nodes[ clientId ].getBoundingClientRect();
-
-		// Create a clientId to Y coördinate map.
-		const clientIdToCoordMap = mapValues( this.nodes, ( node ) =>
-			node.getBoundingClientRect().top - boundaries.top );
-
-		// Cache a Y coördinate to clientId map for use in `onPointerMove`.
-		this.coordMap = invert( clientIdToCoordMap );
-		// Cache an array of the Y coördinates for use in `onPointerMove`.
-		// Sort the coördinates, as `this.nodes` will not necessarily reflect
-		// the current block sequence.
-		this.coordMapKeys = sortBy( Object.values( clientIdToCoordMap ) );
-		this.selectionAtStart = clientId;
+		this.startClientId = clientId;
+		this.props.onStartMultiSelect();
 
-		window.addEventListener( 'mousemove', this.onPointerMove );
-		// Capture scroll on all elements.
-		window.addEventListener( 'scroll', this.onScroll, true );
+		// `onSelectionStart` is called after `mousedown` and `mouseleave`
+		// (from a block). The selection ends when `mouseup` happens anywhere
+		// in the window.
 		window.addEventListener( 'mouseup', this.onSelectionEnd );
+
+		// Removing the contenteditable attributes within the block editor is
+		// essential for selection to work across editable areas. The edible
+		// hosts are removed, allowing selection to be extended outside the
+		// DOM element. `onStartMultiSelect` sets a flag in the store so the
+		// rich text components are updated, but the rerender may happen very
+		// slowly, especially in Safari for the blocks that are asynchonously
+		// rendered. To ensure the browser instantly removes the selection
+		// boundaries, we remove the contenteditable attributes manually.
+		Array.from(
+			this.ref.current.querySelectorAll( '.rich-text' )
```

Comment:
Why  just richText, why not all contenteditables, some blocks might have their own contenteditables/editors.

Reply:
This also feels a bit risky :)

Reply:
It does feel risky. Expanding it to all content editable sounds fine, as long as we are sure that it's reset. I thought rich text only was good to try first.

Reply:
what happens if we don't expand and there's another contenteditable there?

Reply:
You won't be able to select the contents of the `contenteditable` element, but you will be able to select the element as a whole. Tbh I wouldn't expand it to all `contenteditable` elements because we can't disable them in React. So if its re-renders, the element will be editable again. Maybe that's fine because right now it will be editable all the time. But the focus here is on rich text element and allowing those to be partially selected, not on other elements. For those it's fine for now to not be able to select the content.